File: solved.ac/data_structures/1544.js
const input = require("fs")
  .readFileSync("./dev/stdin")
  .toString()
  .trim()
  .split("\n")

const n = +input[0]
const arr = []
let ans = 0
for (let a of input.slice(1)) {
  if (!arr.includes(a)) {
    ans += 1
    for (let i = 0; i < a.length; i++) {
      a = a.slice(1) + a[0]
      if (!arr.includes(a)) arr.push(a)
    }
  }
}

console.log(ans)

File: solved.ac/data_structures/1544.py
result = 0
arr =[]
for _ in range(int(input())):
  word = input()
  if word not in arr:
    result +=1
    for _ in range(len(word)) :
      word = word[1:] + word[0]
      arr.append(word)
print(result)
# import sys;input=sys.stdin.readline
# n = int(input())

# words = {}

# for _ in range(n):
#   word = input().rstrip()
#   print(word)
#   print(word.split())
#   # word = "".join(word.split().sort())
#   # print(word)